File: Cc.Ui/src/models/agency.ts
import { Entity, resource, field, required, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
import { Company } from './company';
import { cache } from '../app/cache';
import { LookupKey } from '../core/lookup';

@resource('agency')
export class Agency extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    companyId: number;
    @field()
    regionId: number;
    @field()
    name: string;
    @field()
    code: string;
    @field()
    provinceCode: string;
    @field()
    cityCode: string;
    @field()
    address: string;
    @field()
    cap: string;
    @field()
    mail: string;
    @field()
    vatCode: string;
    
    @association(Company, 'companyId', (id: string | number) => cache(LookupKey.COMPANY, id))
    company: Company;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Agency {

        let item = new Agency();
        return item;
    }
}

@resource('agency')
export class AgencySearchModel extends Entity implements Pagination {
    
    @field()
    agencyId?: number;
    @field()
    companyId?: number;
    @field()
    regionId?: number;
    @field()
    name?: string;
    @field()
    code?: string;
    @field()
    provinceCode?: string;
    @field()
    cityCode?: string;
    @field()
    address?: string;
    @field()
    cap?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/Agency.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Agency
    {
        public Agency()
        {
            Customer = new HashSet<Customer>();
            CustomerRequiredField = new HashSet<CustomerRequiredField>();
            PrincipalAuth = new HashSet<PrincipalAuth>();
        }

        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public long CompanyId { get; set; }
        public long? RegionId { get; set; }
        public string? Name { get; set; }
        public string? Code { get; set; }
        public string? ProvinceCode { get; set; }
        public string? CityCode { get; set; }
        public string? Address { get; set; }
        public string? Cap { get; set; }
        public string? Mail { get; set; }
        public string? VatCode { get; set; }

        public virtual Company Company { get; set; } = null!;
        public virtual ICollection<Customer> Customer { get; set; }
        public virtual ICollection<CustomerRequiredField> CustomerRequiredField { get; set; }
        public virtual ICollection<PrincipalAuth> PrincipalAuth { get; set; }
    }
}
